libs/cardano-ledger-core/src/Cardano/Ledger/Core/PParams.hs

```diff
@@ -327,7 +329,20 @@ class
   hkdExtraEntropyL :: (HKDFunctor f, ProtVerAtMost era 6) => Lens' (PParamsHKD f era) (HKD f Nonce)
 
   -- | Protocol version
-  hkdProtocolVersionL :: HKDFunctor f => Lens' (PParamsHKD f era) (HKD f ProtVer)
+  hkdProtocolVersionL :: (HKDFunctor f, ProtVerAtMost era 8) => Lens' (PParamsHKD f era) (HKD f ProtVer)
+
+  -- | Protocol version getter
+  hkdProtocolVersionNoUpdateG :: HKDFunctor f => SimpleGetter (PParamsHKD f era) (HKDNoUpdate f ProtVer)
+  default hkdProtocolVersionNoUpdateG :: forall f. (HKDFunctor f, ProtVerAtMost era 8) => SimpleGetter (PParamsHKD f era) (HKDNoUpdate f ProtVer)
+  hkdProtocolVersionNoUpdateG = hkdProtocolVersionL @era @f . Lens.to (toNoUpdate @f @ProtVer)
```

Comment:
I don't think we need this. The only use place for this is in cardano-ledger-pretty, which is not a package I particularly care about, so it should be solved there.

Let's remove this.
```suggestion
```